lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -364,6 +366,37 @@ bool SILPerformanceInliner::isAutoDiffLinearMapWithControlFlow(
   return false;
 }
 
+// Checks if the function may be an AutoDiff VJP returning a partially
+// applied pullback. VJPs return a tuple of the original result and the
+// pullback, which may or may not be partially applied.
+bool SILPerformanceInliner::mayBeAutoDiffVJPReturningPartiallyAppliedPullback(
+    SILValue retVal) {
+  if (auto *ti = dyn_cast<TupleInst>(retVal)) {
+    if (ti->getNumOperands() == 2) {
```

Comment:
Why is specific to VJPs? I think we might consider a generic case: function returns a tuple. If tuple contains a closure, then we increase the benefit.

Reply:
I tried to change things such that the current inlining heuristics remain unchanged and we only make changes specific to autodiff since @eeckstein recommended that it can have many unexpected performance results. Let's wait for him to chime in on this?